client/main.lua

```diff
@@ -76,62 +71,65 @@ local function areKeysJobShared(veh)
+                        TriggerServerEvent('qb-vehiclekeys:server:AcquireVehicleKeys', vehPlate)
                     end
+
                     return true
                 end
             end
         end
     end
+
     return false
 end
 
-local function setVehicleDoorLock(veh, state, anim)
-    if veh then
-        if not isBlacklistedVehicle(veh) then
-            if hasKeys(qbx.getVehiclePlate(veh)) or areKeysJobShared(veh) then
-                local vehLockStatus = GetVehicleDoorLockStatus(veh)
+---manages the opening of locks
+---@param vehicle number? The entity number of the vehicle.
+---@param state boolean? State of the vehicle lock.
+---@param anim any Aniation
+local function setVehicleDoorLock(vehicle, state, anim)
+    if not vehicle then return end
+    if not isBlacklistedVehicle(vehicle) then
+        if hasKeys(qbx.getVehiclePlate(vehicle)) or areKeysJobShared(vehicle) then
+
+            if anim then
+                lib.requestAnimDict('anim@mp_player_intmenu@key_fob@')
+                TaskPlayAnim(cache.ped, 'anim@mp_player_intmenu@key_fob@', 'fob_click', 3.0, 3.0, -1, 49, 0, false, false, false)
+            end
 
-                if anim then
-                    lib.requestAnimDict('anim@mp_player_intmenu@key_fob@')
-                    TaskPlayAnim(cache.ped, 'anim@mp_player_intmenu@key_fob@', 'fob_click', 3.0, 3.0, -1, 49, 0, false, false, false)
-                end
+            TriggerServerEvent('InteractSound_SV:PlayWithinDistance', 5, 'lock', 0.3)
+            NetworkRequestControlOfEntity(vehicle)
 
-                TriggerServerEvent("InteractSound_SV:PlayWithinDistance", 5, "lock", 0.3)
-
-                NetworkRequestControlOfEntity(veh)
-                if state then
-                    state = state == true and 2 or 1
-                    TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(veh), state)
-                    exports.qbx_core:Notify(state == 2 and locale("notify.vehicle_locked") or locale("notify.vehicle_unlocked"), 'inform')
-                else
-                    if vehLockStatus == 1 then
-                        TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(veh), 2)
-                        exports.qbx_core:Notify(locale("notify.vehicle_locked"), 'inform')
-                    else
-                        TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(veh), 1)
-                        exports.qbx_core:Notify(locale("notify.vehicle_unlocked"), 'inform')
-                    end
-                end
-                SetVehicleLights(veh, 2)
-                Wait(250)
-                SetVehicleLights(veh, 1)
-                Wait(200)
-                SetVehicleLights(veh, 0)
-                Wait(300)
-                ClearPedTasks(cache.ped)
+            local lockstate
+            if state then
+                lockstate = state == true and 2 or 1
```

Comment:
The `== true` part can be removed, it will function the same

Reply:
```lua
            if state then
                lockstate = state == true and 2 or 1
```

Reply:
The `== true`  improves readability

Reply:
That is how we write all of our ternary operators

Reply:
I already noticed what the problem was